internal/pkg/pkg.go

```diff
@@ -39,6 +39,10 @@ import (
 const CurDir = "."
 const ParentDir = ".."
 
+const (
+	pkgPathAnnotation = "internal.config.kubernetes.io/package-path"
```

Comment:
I like the name in the PR description: "orchestrator.config"

Let's go tentatively with this, may still change based on upstream spec change.

Reply:
Ack.
Actually I started with `orchestrator.config.kubernetes.io` and then later changed to this annotation. I agree `orchestrator.config.kubernetes.io` is more accurate here.

Reply:
Is this related to https://github.com/kubernetes-sigs/kustomize/pull/3995#discussion_r655646335?

Reply:
This should be added to `kioutil` package in kyaml for functions to consume.

Here are the series of steps:
1. Add this to kioutil package and release kyaml
2. Reference it from this PR.
3. Update [upsert-resource](https://github.com/GoogleContainerTools/kpt-functions-catalog/blob/master/functions/go/upsert-resource/upsertresource/upsert_resource.go#L120) function to retain this annotation as well. 
4. Release kpt and upsert-resource.

It is a bit tedious process.

Reply:
Please CC Katrina in the kyaml PR

Reply:
Functions shouldn't consume this, that's the point of `internal` annotation.

Reply:
This annotation will be dropped by `upsert-resource` function then.

Reply:
See https://github.com/kubernetes-sigs/kustomize/pull/3995

Functions SHOULD NOT mutate annotations with prefix `internal.config.kubernetes.io`. So the upsert function should ensure this is preserved.

Reply:
kyaml will provide utility for doing that

Reply:
@phanimarupaka As Frank indicated, this is a control annotation (specific to kpt in this case, no need to move to kyaml), function should not mutate these annotations.

Orchestrator (`kpt fn render`) also adds this annotation if it doesn't exist on a resource post function execution (to take care of generator use-case). So, I think, in this case if upsert drops this annotation on resources, it will be added back. 

Another note: `fn eval` doesn't uses this annotation because it's not package aware.

Are people suppose to use `upsert-resource` in declarative flow or imperative flow.

Reply:
upsert-resource should be used as part of imperative flow. Users might choose to use it as part of declarative flow as well but it doesn't make much sense.

I am fine with it if it is added back if not present.